src/solver/getitem/backward_getitem.cpp

```diff
@@ -0,0 +1,279 @@
+
+        kernel.l_wk.push_back(xlocalsize);
+        kernel.l_wk.push_back(ylocalsize);
+        kernel.l_wk.push_back(zlocalsize);
+
+        kernel.g_wk.push_back(xgridsize);
+        kernel.g_wk.push_back(ygridsize);
+        kernel.g_wk.push_back(zgridsize);
+
+        result.construction_params.push_back(kernel);
+    }
+
+    {
+        size_t xlocalsize = LOCAL_SIZE;
+        size_t xgridsize  = AlignUp(dy_numel, xlocalsize);
+        size_t ylocalsize = 1;
+        size_t ygridsize  = 1;
+        size_t zlocalsize = 1;
+        size_t zgridsize  = 1;
+
+        auto kernel = KernelInfo{};
+
+        kernel.kernel_file = "MIOpenGetitem.cpp";
+        kernel.kernel_name = "GetitemBwd";
+
+        const auto build_params = KernelBuildParameters{
+            {"MIOPEN_USE_FP16", static_cast<int>(dtype == miopenHalf)},
+            {"MIOPEN_USE_FP32", static_cast<int>(dtype == miopenFloat)},
+            {"MIOPEN_USE_BFP16", static_cast<int>(dtype == miopenBFloat16)},
+            {"INPUT_TYPE", input_dtype == "bfloat16" ? "ushort" : input_dtype},
+            {"INDEX_TYPE", index_dtype},
+            {"ERROR_TYPE", error_dtype},
+            {"OUTPUT_TYPE", output_dtype == "bfloat16" ? "ushort" : output_dtype},
+            {"LOCAL_SIZE", LOCAL_SIZE},
+        };
+
+        kernel.comp_options = build_params.GenerateFor(kbp::HIP{});
+
+        kernel.l_wk.push_back(xlocalsize);
+        kernel.l_wk.push_back(ylocalsize);
+        kernel.l_wk.push_back(zlocalsize);
+
+        kernel.g_wk.push_back(xgridsize);
+        kernel.g_wk.push_back(ygridsize);
+        kernel.g_wk.push_back(zgridsize);
+
+        result.construction_params.push_back(kernel);
+    }
+
+    result.invoker_factory = [](const std::vector<Kernel>& kernels) {
+        return [=](const Handle& handle_, const AnyInvokeParams& raw_params) {
+            decltype(auto) params = raw_params.CastTo<miopen::getitem::GetitemInvokeParams>();
+
+            auto start_dim = params.dims[0];
+            auto dx_dims   = params.dxDesc.GetLengths();
+
+            auto dims     = params.dims;
+            auto dimCount = params.dimCount;
+
+            std::vector<int32_t> output_dims(dimCount);
+            for(int32_t i = 0; i < dimCount; i++)
+            {
+                output_dims[i] = static_cast<int32_t>(dx_dims[dims[i]]);
+            }
+
+            auto indexCount = params.indexCount;
+            auto index_dims = params.indexDescs[0]->GetLengths();
+            auto sliceCount = params.sliceCount;
+            auto slices     = params.slices;
+            auto dim_info_offset =
+                indexCount > 0 ? indexCount * static_cast<int32_t>(index_dims[0]) : 0;
+
+            auto dy_tv = get_inner_expanded_tv<5>(params.dyDesc);
+            auto dx_tv = get_inner_expanded_tv<5>(params.dxDesc);
+
+            slice_tv<5>(dx_tv, sliceCount, slices);
+
+            auto elapsed = 0.f;
+            HipEventPtr start;
+            HipEventPtr stop;
```

Comment:
Since multiple kernels are called here, the kernel gap is measured by subtracting the time when the first kernel is called from the time when the last kernel ends.